src/x509.c

```diff
@@ -7517,6 +7517,464 @@ WOLFSSL_X509_CRL* wolfSSL_d2i_X509_CRL(WOLFSSL_X509_CRL** crl,
+    if (wolfSSL_BIO_write(bio, "        CRL extensions:\n",
+                  (int)XSTRLEN("        CRL extensions:\n")) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+
+    #if !defined(NO_SKID)
+    if (crl->crlList->extAuthKeyIdSet && crl->crlList->extAuthKeyId != NULL) {
+        char tmp[100];
+        word32 i;
+        char val[5];
+        int valSz = 5;
+        int len = 0;
+
+        if (XSNPRINTF(tmp, sizeof(tmp),
+                  "            X509v3 Authority Key Identifier: ")
+            >= (int)sizeof(tmp))
+        {
+            WOLFSSL_MSG("buffer overrun");
+            return WOLFSSL_FAILURE;
+        }
+        XSTRNCAT(tmp, "\n", sizeof(tmp) - XSTRLEN(tmp) - 1);
+
+        if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+        XMEMSET(tmp, 0, sizeof(tmp));
+
+        if (XSNPRINTF(tmp, sizeof(tmp), "                keyid")
+            >= (int)sizeof(tmp))
+        {
+            WOLFSSL_MSG("buffer overrun");
+            return WOLFSSL_FAILURE;
+        }
+        for (i = 0; i < XSTRLEN((char*)crl->crlList->extAuthKeyId); i++) {
+            /* check if buffer is almost full */
+            if (XSTRLEN(tmp) >= sizeof(tmp) - valSz) {
+                if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+                    return WOLFSSL_FAILURE;
+                }
+                tmp[0] = '\0';
+            }
+            if (XSNPRINTF(val, valSz, ":%02X", crl->crlList->extAuthKeyId[i])
+                >= valSz)
+            {
+                WOLFSSL_MSG("buffer overrun");
+                return WOLFSSL_FAILURE;
+            }
+            XSTRNCAT(tmp, val, valSz);
+        }
+        len = (int)XSTRLEN("\n");
+        XSTRNCAT(tmp, "\n", len + 1);
+        if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+    }
+    #endif
+
+    if (crl->crlList->crlNumber) {
+        char tmp[20];
+ 
+        if (wolfSSL_BIO_write(bio, "            X509v3 CRL Number:\n",
+                      (int)XSTRLEN("            X509v3 CRL Number:\n")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+
+        if (XSNPRINTF(tmp, sizeof(tmp), "                %d\n",
+            crl->crlList->crlNumber) >= (int)sizeof(tmp))
+        {
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+
+    }
+
+    if (crl->crlList->totalCerts > 0) {
+        char tmp[80];
+        RevokedCert* revoked = crl->crlList->certs;
+
+        if (wolfSSL_BIO_write(bio, "Revoked Certificates:\n",
+                       (int)XSTRLEN("Revoked Certificates:\n")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+
+        for (int i = 0; i < crl->crlList->totalCerts; i++) {
+            if (revoked->serialSz > 0) {            
+                unsigned char serial[32];
+                int  sz = sizeof(serial);
```

Comment:
Add cast to `int`. You can catch these by using `CC=g++`.

Reply:
Fixed. Thanks for the tip